tests/test_remote.py

```diff
@@ -0,0 +1,225 @@
+        """Tests that samples are processed by the samples_expectation SF
+        function by using a mocked function instead which returns a pre-defined
+        value."""
+        modes = 8
+        shots = 10
+        dev = qml.device('strawberryfields.ai', backend="X8", wires=modes, shots=shots)
+
+        @qml.qnode(dev)
+        def quantum_function(theta, phi):
+            qml.Beamsplitter(theta, phi, wires=[0,1])
+            qml.Beamsplitter(theta, phi, wires=[4,5])
+            return qml.expval(qml.TensorN(wires=list(range(modes))))
+
+        expected_expval = 100
+
+        monkeypatch.setattr("strawberryfields.RemoteEngine", MockEngine)
+        monkeypatch.setattr("pennylane_sf.remote.samples_expectation", lambda *args: expected_expval)
+        a = quantum_function(1., 0)
+
+        assert a == expected_expval
+class TestVariance:
+    """Test that variances are correctly returned from the hardware device."""
+
+    def test_mocked_engine_run_var(self, monkeypatch):
+        """Tests that samples are processed by the samples_variance SF function
+        by using a mocked function instead which returns a pre-defined value."""
+        modes = 8
+        shots = 10
+        dev = qml.device('strawberryfields.ai', backend="X8", wires=modes, shots=shots)
+
+        @qml.qnode(dev)
+        def quantum_function(theta, phi):
+            qml.Beamsplitter(theta, phi, wires=[0,1])
+            qml.Beamsplitter(theta, phi, wires=[4,5])
+            return qml.var(qml.TensorN(wires=list(range(modes))))
+
+        expected_var = 100
+
+        monkeypatch.setattr("strawberryfields.RemoteEngine", MockEngine)
+        monkeypatch.setattr("pennylane_sf.remote.samples_variance", lambda *args: expected_var)
+        var = quantum_function(1., 0)
+
+        assert var == expected_var
+
+
+class TestProbs:
+    """Test that probabilities are correctly returned from the hardware device."""
+
+    def test_mocked_engine_run_all_fock_probs(self, monkeypatch, tol):
+        """Tests that probabilities are correctly summed when specifying a
+        subset of the wires and using a mock SF RemoteEngine."""
+        modes = 8
+        shots = 10
+        dev = qml.device('strawberryfields.ai', backend="X8", wires=modes, shots=shots)
+
+        @qml.qnode(dev)
+        def quantum_function(theta, phi):
+            qml.Beamsplitter(theta, phi, wires=[0,1])
+            qml.Beamsplitter(theta, phi, wires=[4,5])
+            return qml.probs(wires=[0, 1])
+
+        expected_probs = np.array([[0. , 0. , 0. , 0. , 0. ],
+                                   [0.1, 0. , 0.3, 0. , 0. ],
+                                   [0. , 0.1, 0. , 0.1, 0. ],
+                                   [0. , 0. , 0. , 0. , 0.1],
+                                   [0. , 0.1, 0.1, 0.1, 0. ]])
+
+        monkeypatch.setattr("strawberryfields.RemoteEngine", MockEngine)
+        probs = quantum_function(1., 0)
+
+        assert np.allclose(probs, expected_probs, atol=tol)
```

Comment:
It might also be good to explicitly specify `cutoff` in this test, to make sure that the returned expected probs are of the right shape

Reply:
Good idea! Added a check to see that the shape of `probs` matches the cutoff used internally by `all_fock_probs`